src/main/webapp/app/exercises/shared/dashboards/tutor/exercise-assessment-dashboard.component.ts

```diff
@@ -144,7 +144,7 @@ export class ExerciseAssessmentDashboardComponent implements OnInit, AfterViewIn
     ngOnInit(): void {
         this.exerciseId = Number(this.route.snapshot.paramMap.get('exerciseId'));
         this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));
-        this.isTestRun = this.route.snapshot.url[3]?.toString() === 'test-run-exercise-assessment-dashboard';
+        this.isTestRun = this.route.snapshot.url.length > 8 && this.route.snapshot.url[8]?.toString() === 'test-run-exercise-assessment-dashboard';
```

Comment:
I don't know how but can this be done without just writing 8 there?

Reply:
By using the router perhaps. I didn't do much here to reduce the risk of introducing regressions. Tbh, if we want to change this we should probably think of a way to determine if this is a test run without relying on some magic values in the url.